LedgerHQ/ledger-live
# Feat/live 4839 anon wallet account

### 📝 Description

Anonymise ledger live account ids before sending them to an eWApp (External Wallet App).

This anonymisation has been done on the new `wallet-api` part as well as the legacy  / _soon to be deprecated_ `platform` part.

Note: account ids are not anonymised on the `platform` side since some actions (such as `transaction.sign`, `transaction.broadcast` or `exchange.complete`) rely on Ledger Live `Operation` type that use ledger live account ids.

These actions could be updated / refactored to exclude Ledger Live account ids. But if this code is supposed to be deprecated soon and replaced by the `wallet-api` version (that don't have such issues since it does not rely on Ledger Live `Operation` type), this might not be worth the effort (and the risk of breaking these flows).

### ❓ Context

- **Impacted projects**: `LLC`
- **Linked resource(s)**: https://ledgerhq.atlassian.net/browse/LIVE-4839

### ✅ Checklist


### 📸 Demo



### 🚀 Expectations to reach

Manually tested using:
- Production debug app for the V1 (`platform/` part)
- https://github.com/LedgerHQ/platform-app-debug/pull/4 branch for V2 (`wallet-api` part)

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._
